.goreleaser.yml

```diff
@@ -1,5 +1,19 @@
 before:
   hooks:
+    # The build is done in this particular way to build xcaddy in a designated directory named in .gitignore.
+    # This is so we can run goreleaser on tag without Git complaining of being dirty. The main.go in cmd/xcaddy directory 
+    # cannot be built within that directory due to changes necessary for the build causing Git to be dirty, which
+    # subsequently causes gorleaser to refuse running.
+    - mkdir -p xcaddy-build
```

Comment:
GitHub isn't allowing me to comment on a line that's not part of the diff, but line 24 below needs to be changed to `dir: ./xcaddy-build`.